plugin.php

```diff
@@ -289,9 +289,12 @@ function rest_validate_request_arg( $value, $request, $param ) {
 		$args = $attributes['args'][ $param ];
 
 		if ( ! empty( $args['enum'] ) ) {
-			if ( ! in_array( $value, $args['enum'] ) ) {
+			if ( 'string' === $args['type'] && ! in_array( $value, $args['enum'] ) ) {
 				return new WP_Error( 'rest_invalid_param', sprintf( __( '%s is not one of %s' ), $param, implode( ', ', $args['enum'] ) ) );
 			}
+			if ( 'array' === $args['type'] && array_diff( $value, $args['enum'] ) ) {
```

Comment:
Oh. I think I introduced this because `roles` ended up with an enum in my merge conflict:

`
'roles'           => array(
	'description' => __( 'Roles assigned to the resource.' ),
	'type'        => 'array',
	'context'     => array( 'edit' ),
	'enum'        => array_keys( $wp_roles->role_objects ),
),
```


I'll revert for now.